File: dart/basics/1480.running-sum-of-1-d-array.dart
/*
 * @lc app=leetcode id=1480 lang=dart
 *
 * [1480] Running Sum of 1d Array
 */

// @lc code=start
class Solution {
  List<int> runningSum(List<int> nums) {
    List<int> rs = [];
    int newNum = 0;
    for (int num in nums) {
      newNum += num;
      rs.add(newNum);
    }
    return rs;
  }
}

// @lc code=end

File: java/basics/1480.running-sum-of-1-d-array.java
package basics;
/*
 * @lc app=leetcode id=1480 lang=java
 *
 * [1480] Running Sum of 1d Array
 */

// @lc code=start
class Solution {
    public int[] runningSum(int[] nums) {
        int[] runningSum = new int[nums.length];
        for (int i = 0; i < nums.length; i++) {
            runningSum[i] = i == 0 ? nums[i] : runningSum[i - 1] + nums[i];
        }
        return runningSum;
    }
}
// @lc code=end
